fpga_interchange/arch.cc

```diff
@@ -192,6 +193,50 @@ Arch::Arch(ArchArgs args) : args(args)
         }
     }
 
+    // Initially LutElement vectors for each tile type.
+    tile_type_index = 0;
+    lut_elements.resize(chip_info->tile_types.size());
+    for (const TileTypeInfoPOD &tile_type : chip_info->tile_types) {
+        std::vector<LutElement> &elements = lut_elements[tile_type_index++];
+        elements.reserve(tile_type.lut_elements.size());
+        for (auto &lut_element : tile_type.lut_elements) {
+            elements.emplace_back();
+
+            LutElement &element = elements.back();
+            element.width = lut_element.width;
+            for (auto &lut_bel : lut_element.lut_bels) {
+                auto result = element.lut_bels.emplace(IdString(lut_bel.name), LutBel());
+                NPNR_ASSERT(result.second);
+                LutBel &lut = result.first->second;
+
+                lut.low_bit = lut_bel.low_bit;
+                lut.high_bit = lut_bel.high_bit;
+
+                lut.pins.reserve(lut_bel.pins.size());
+                for (size_t i = 0; i < lut_bel.pins.size(); ++i) {
+                    IdString pin(lut_bel.pins[i]);
+                    lut.pins.push_back(pin);
+                    lut.pin_to_index[pin] = i;
+                }
+            }
+
+            element.compute_pin_order();
+        }
+    }
+
+    // Map lut cell types to their LutCellPOD
+    for (const LutCellPOD &lut_cell : chip_info->cell_map->lut_cells) {
+        IdString cell_type(lut_cell.cell);
+        auto result = lut_cells.emplace(cell_type, &lut_cell);
+        NPNR_ASSERT(result.second);
+    }
+
+    raw_bin_constant = std::regex("[01]+", std::regex_constants::ECMAScript | std::regex_constants::optimize);
+    verilog_bin_constant =
```

Comment:
I appreciate the priority is getting something working right now; but this kind of stuff is why I'd like to see something other than Verilog-constants-in-strings as the bitvector format long term.

Reply:
Changing that will require patching up https://github.com/Xilinx/RapidWright/blob/4ed91eab267c4476e4e2318f89864bc3d63cae69/src/com/xilinx/rapidwright/interchange/LogNetlistReader.java#L57-L80 .  I'm not saying that we shouldn't change it, just that there are couple moving parts here.  Because we want to stand up a direct FPGA interchange to FASM emitter (https://github.com/SymbiFlow/python-fpga-interchange/issues/27) we should probably cut over to something sooner rather than later.

It is worth noting that right now it appears that all bitstrings are coming into the LogicalNetlist as char vectors of `1` and `0` (e.g. "raw_bin_constant").  This appears to be accepted by RapidWright/Vivado for LUTs.  I am expecting this will be a problem for RAMB18E1 init strings.